tests/integration/utils.py

```diff
@@ -463,3 +472,170 @@ def _validate_expected_dep_file_contents(dep_contents_file: Path, output_dir: Pa
         dep_file = output_dir / "deps" / path
         assert dep_file.exists()
         assert dep_file.read_text() == expected_content
+
+
+def retrieve_changed_files_from_git() -> tuple[Path, ...]:
+    repo = Repo(".", search_parent_directories=True)
+    # >>> type(repo.branches)
+    # <class 'git.util.IterableList'>
+    # Despite the fact stated above mypy does not believe one can use 'in' with
+    # repo.branches because branches is an alias to heads and heads are decorated
+    # with @property:
+    #  Unsupported right operand type for in ("Callable[[], IterableList[Head]]")
+    main = "main" if "main" in repo.branches else "origin/main"  # type: ignore
+    try:
+        files = repo.git.diff("--name-only", f"{main}..HEAD").split("\n")
+    # Widest net possible in order not to interfere with testing for no good reason.
+    except Exception:
+        # If there is no main and no origin/main then someone is probably doing
```

Comment:
Please log this!

Reply:
Good point, done.

Reply:
When I log something anywhere within these functions, I see nothing unless I add `--log-cli-level=<level>` in `[testenv:integration]`. But the option is already configured below in the [file](https://github.com/containerbuildsystem/cachi2/blob/main/tox.ini#L77)

Is it just me? I don't understand.

Reply:
> Is it just me? 

It's not, turns out, we need to set `log_cli=true` for pytest as well then I can see:

`2024-09-13 17:44:38 DEBUG HELLO FROM TEST`

based on this diff:
```
diff --git a/tests/integration/utils.py b/tests/integration/utils.py
index 80db1fd1..f7c89398 100644
--- a/tests/integration/utils.py
+++ b/tests/integration/utils.py
@@ -484,6 +484,7 @@ def retrieve_changed_files_from_git() -> tuple[Path, ...]:
     #  Unsupported right operand type for in ("Callable[[], IterableList[Head]]")
     main = "main" if "main" in repo.branches else "origin/main"  # type: ignore
     try:
+        log.debug("HELLO FROM TEST")
         files = repo.git.diff("--name-only", f"{main}..HEAD").split("\n")
     # Widest net possible in order not to interfere with testing for no good reason.
     except Exception as e:
```

And I've even seen @a-ovchinnikov's warning when I made a typo in there :D:
```
------------------------------------------ live log collection -------------------------------------------
2024-09-13 17:48:10 WARNING Detection of changed files unexpectedly failed. This either indicates that both 'main' and 'origin/main' branches are missing in a repo or a more fundamental failure. The tool will attempt to recover from this by not filtering any tests out. This is the exception that was encountered: 'Logger' object has no attribute 'DEBUG'
collected 52 items
```

so yeah, `log_cli=true` works

Reply:
Yeah that would work, but then `tox -e py3` shows countless amount of logs.

Reply:
`log_cli_level = DEBUG` is supposed to log everything. it should be at least `INFO` for tests, or maybe even just `WARNING`.

Reply:
> Yeah that would work, but then tox -e py3 shows countless amount of logs 🤯

It does, do you have a better idea how to handle debug logs from test? I mean, it's kinda expected that once we turn it on, debug logs litter the output, that's their inherent nature, isn't it?

Neither is IMO a problem this PR inherently needs to solve (unless you favour doing it here). We identified the problem and I agree we should decrease the default log level, allowing users to override it from tox's CLI if needed.

Reply:
I thought that we could use `log_cli` only for integration tests, but `log-cli` is not available in CLI.
Let's drop it in this PR.